src/function_wrappers.ts

```diff
@@ -122,6 +131,37 @@ const wrapHttpFunction = (execute: HttpFunction): RequestHandler => {
   };
 };
 
+const wrapHttpAsyncFunction = (
+  userFunction: OpenFunction,
+  context: OpenFunctionContext
+): RequestHandler => {
+  const ctx = OpenFunctionRuntime.ProxyContext(context);
+  const httpHandler = (req: Request, res: Response) => {
+    const callback = getOnDoneCallback(res);
+
+    Promise.resolve()
+      .then(() => userFunction(ctx, req.body))
+      .then(result => {
+        debug('ℹ️ User function returned: %j', result);
+
+        const data = get(result, 'body');
+        const code = get(result, 'code', 200);
+        const headers = get(result, 'headers');
+
+        !isEmpty(headers) && res.set(headers);
+
+        if (data !== undefined) {
+          res.status(code).send(data);
```

Comment:
I think wrapHttpAsyncFunction can be changed to wrapOpenFunction because when there're no outputs, userFunction is just a sync function, userFunction handles all the input data and send response to the caller. When there're outputs, userFunction is an async function.

This way the sync and async function has the same signature actually. and we call this signature OpenFunction

Reply:
Make sense, will make this change.